zebra-network/src/meta_addr.rs

```diff
@@ -656,32 +739,83 @@ impl Ord for MetaAddr {
         use std::net::IpAddr::{V4, V6};
         use Ordering::*;
 
-        let oldest_first = self.get_last_seen().cmp(&other.get_last_seen());
-        let newest_first = oldest_first.reverse();
-
-        let connection_state = self.last_connection_state.cmp(&other.last_connection_state);
-        let reconnection_time = match self.last_connection_state {
-            Responded => oldest_first,
-            NeverAttemptedGossiped => newest_first,
-            NeverAttemptedAlternate => newest_first,
-            Failed => oldest_first,
-            AttemptPending => oldest_first,
-        };
-        let ip_numeric = match (self.addr.ip(), other.addr.ip()) {
+        // First, try states that are more likely to work
+        let responded_never_failed_attempting =
+            self.last_connection_state.cmp(&other.last_connection_state);
+
+        // # Security and Correctness
+        //
+        // Prioritise older attempt times, so we try all peers in each state,
+        // before re-trying any of them. This avoids repeatedly reconnecting to
+        // peers that aren't working.
```

Comment:
This is a core part of the security fix: try older peers first, to reduce rapid reconnections to the same peer.